editors/vscode/client/extension.ts

```diff
@@ -90,10 +92,38 @@ export async function activate(context: ExtensionContext) {
   const outputChannel = window.createOutputChannel(outputChannelName);
   const traceOutputChannel = window.createOutputChannel(traceOutputChannelName);
 
-  const ext = process.platform === 'win32' ? '.exe' : '';
-  // NOTE: The `./target/release` path is aligned with the path defined in .github/workflows/release_vscode.yml
-  const command = process.env.SERVER_PATH_DEV ??
-    join(context.extensionPath, `./target/release/oxc_language_server${ext}`);
+  function findBinary(): string {
+    const cfg = workspace.getConfiguration('oxc');
+
+    let bin = cfg.get<string>('binPath', '');
+    if (bin && existsSync(bin)) {
+      return bin;
+    }
+
+    const workspaceFolders = workspace.workspaceFolders;
+    if (workspaceFolders) {
+      for (const folder of workspaceFolders) {
+        const binPath = join(
+          folder.uri.fsPath,
+          'node_modules',
+          '.bin',
+          'oxc_language_server',
+        );
+        if (existsSync(binPath)) {
```

Comment:
issue: synchronous FS api in a loop. Use `Promise.all` +  `fs.promises.exists` instead.

Reply:
Perhaps we can use `Promise.any`, but this requires `ES2021`. What do you think?

Reply:
es2021 is fine imo, it's been supported since node `15.14.0` which is > 3 yrs old. if anyone is still using a vscode version that's that old, they shuold just update